From 8afd39abc901becf02f338b176b47cd0635bd137 Mon Sep 17 00:00:00 2001
From: Eric Sauvageau <merlin@asuswrt-merlin.net>
Date: Sun, 9 Aug 2026 17:00:05 -0400
Subject: [PATCH] shared: fix potential buffer overruns in remove_dups()

Based in part on the Reaper fix, also protect the final string copy.

(cherry picked from commit e1d0918e314cf44e5a14f815f5b1018071c3360e)
---
 release/src/router/shared/shutils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/release/src/router/shared/shutils.c b/release/src/router/shared/shutils.c
index 52fb19c7ee..4485d0dd4d 100644
--- a/release/src/router/shared/shutils.c
+++ b/release/src/router/shared/shutils.c
@@ -1521,7 +1521,7 @@ remove_dups(char *inlist, int inlist_size)
 		}
 	}
 
-	strncpy(inlist, outlist, inlist_size);
+	strlcpy(inlist, outlist, inlist_size);
 
 	free(outlist);
 	return inlist;
-- 
2.25.1

